tests/factory/test_factory_scraper.py

```diff
@@ -2,88 +2,62 @@
-        'fname': '2013-10-01-03-02-04-mozilla-central-firefox-27.0a1.en-US.win32.installer.exe',
-        'kwargs': {
-            'platform': 'win32',
-        },
-    },
+    ('daily', None,
+     '2013-10-01-03-02-04-mozilla-central-firefox-27.0a1.en-US.win32.installer.exe', {
+         'platform': 'win32',
+     }),
 
     # TinderboxScraper
-    {
-        'scraper_type': 'tinderbox',
-        'fname': '1374583608-mozilla-central-firefox-25.0a1.en-US.win32.installer.exe',
-        'kwargs': {
-            'platform': 'win32',
-        },
-    },
+    ('tinderbox', None, '1374583608-mozilla-central-firefox-25.0a1.en-US.win32.installer.exe', {
+        'platform': 'win32',
+    }),
 
     # TryScraper
-    {
-        'scraper_type': 'try',
-        'builds': ['/firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/try-foobar/'],
-        'fname': '8fcac92cfcad-firefox-38.0a1.en-US.mac.dmg',
-        'kwargs': {
-            'revision': '8fcac92cfcad',
-            'platform': 'mac64',
-        },
-    },
-]
-
-
-class TestFactoryCorrectScraper(mhttpd.MozHttpdBaseTest):
-    """Test mozdownload for the factory scraper."""
-
-    @patch('mozdownload.treeherder.Treeherder.query_builds_by_revision')
-    def test_factory(self, query_builds_by_revision):
-        """Testing various download scenarios for the factory."""
-
-        base_kwargs = {
-            'base_url': self.wdir,
-            'destination': self.temp_dir,
-            'log_level': 'ERROR',
-        }
-
-        for test in tests:
-            if test.get('builds'):
-                query_builds_by_revision.return_value = test['builds']
-
-            kwargs = copy.deepcopy(base_kwargs)
-            kwargs.update(test.get('kwargs'))
-
-            build = FactoryScraper(test['scraper_type'], **kwargs)
-            build.download()
-
-            dir_content = os.listdir(self.temp_dir)
-            self.assertTrue(test['fname'] in dir_content)
-
-            mozfile.remove(os.path.join(self.temp_dir, test['fname']))
+    ('try', ['/firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/try-foobar/'],
+     '8fcac92cfcad-firefox-38.0a1.en-US.mac.dmg', {
+         'revision': '8fcac92cfcad',
+         'platform': 'mac64',
+     }),
+])
+def test_factory(httpd, tmpdir, scraper_type, builds, fname, arg, mocker):
+    """Testing various download scenarios for the factory."""
+    query_builds_by_revision \
+        = mocker.patch('mozdownload.treeherder.Treeherder.query_builds_by_revision')
+    base_kwargs = {
+        'base_url': httpd.get_url(),
+        'destination': str(tmpdir),
+        'log_level': 'ERROR',
+    }
+
+    if builds:
+        query_builds_by_revision.return_value = builds
+
+    kwargs = copy.deepcopy(base_kwargs)
+    kwargs.update(arg)
+
+    build = FactoryScraper(scraper_type, **kwargs)
+    build.download()
+
+    dir_content = os.listdir(str(tmpdir))
+    assert fname in dir_content
+
+    mozfile.remove(os.path.join(str(tmpdir), fname))
```

Comment:
The file is in the temporary directory and as such will automatically be removed when the `tmpdir` fixture runs its clean-up steps. Also we want to get rid of the `mozfile` dependency.